precommit/src/main/shell/core.d/00-yetuslib.sh

```diff
@@ -528,3 +528,65 @@ function yetus_set_trap_handler
     trap "${func} ${signal}" "${signal}"
   done
 }
+
+## @description  Determine if running in a container
+## @audience     public
+## @stability    evolving
+## @replaceable  no
+function yetus_is_container
+{
+  declare mounts
+  declare cgroups
+
+  if [[ -n "${YETUS_CONTAINER_STATE}" ]]; then
+    if [[ "${YETUS_CONTAINER_STATE}" == "true" ]]; then
+      return 0
+    fi
+    return 1
+  fi
+
+  if [[ -f /.dockerenv ]]; then
+    YETUS_CONTAINER_STATE=true
+    return 0
+  fi
+
+  if [[ -n "${container}" ]]; then
```

Comment:
Where does `${container}` come from?

Reply:
From the reading I've done, lxc will set `${container}` to `lxc` and some other container technologies will also set that to ... something.  It's a terrible check, really, but what else can we do? :shrug:

Reply:
I've added some comments to this code block to hopefully better describe what is going on.